osquery/main/windows/main.cpp

```diff
@@ -393,14 +316,26 @@ int main(int argc, char* argv[]) {
        static_cast<LPSERVICE_MAIN_FUNCTION>(osquery::ServiceMain)},
       {nullptr, nullptr}};
 
+  auto shutdown = ([]() {
+    // The event only gets initialized in the entry point of the service. Child
+    // processes and those run from the commandline will have kStopEvent as a
+    // nullptr
+    if (osquery::kStopEvent != nullptr) {
```

Comment:
Looks like the problem is that this never hits for either worker or watcher processes as we don't have inheritance enabled for the `kStopEvent` event. I have a [branch here](https://github.com/muffins/osquery/tree/win-remove-osqueryd) that's nearly got this working, but the issue now (I believe) is that this shutdown is only ever registered for the child workers, but the watcher process never receives the event notification that it itself needs to shut down. Will continue investigating this in the morning, but I think we're nearly there.